### PR TITLE
USB: Use default VID/PID for RIOT-included peripherals

### PR DESCRIPTION
**Pull request by chrysn**
### Contribution description

This adds a check for whether any custom user code (ie. anything but built-in peripherals) uses USB, and if nothing does, allows for 1209/7D00 to be the default VID/PID pair.

It follows the action plan of https://github.com/RIOT-OS/RIOT/issues/12273#issuecomment-575257426, with the simplification that it operates per compilation unit, and thus the ifdef/define combination can be inside the include guard where those things usually are.

### Alternatives

If the build system can tell RIOT's compilation units apart from application compilation units or packages, it could set USB_H_USER_IS_RIOT on all modules. That would significantly reduce the number of files this touches. (But then again, any helpers for creating custom HID devices would need to unset the flag again, so meh).

The name USB_H_USER_IS_RIOT is up for bikeshedding.

### Testing procedure

* Pick a board with USB support (eg. samr21-xpro).
* In any example that does not do anything USB-related, add the `usbus_cdc_ecm`, `auto_init_usbus` and `auto_init_gnrc_netif` modules without setting CONFIG_USB_VID. (`stdio_cdc_acm` would work the same).
* The example builds fine, and now has an additional Ethernet interface.
* For comparison, take an example that does anything custom USB, like usbus_uart_adapter from https://github.com/RIOT-OS/RIOT/pull/12268
* In that example, remove the CONFIG_USB_PID settings (incidentally, if you right now merge master into that PR, currently that is already the case as the renaming of USB_CONFIG_PID to CONFIG_USB_PID is not in there yet).
* Building this will result in an error message that you need to set a VID/PID, and recommends the 1209/7D01 pair reserved for RIOT demos, tests and experiments.

### Issues/PRs references

This is a first step in properly addressing https://github.com/RIOT-OS/RIOT/issues/12273.

A follow-up (preferably in an own PR, but if you prefer I can pull it in here as well) should move the usb_id_check that's currently replicated across examples into common make infrastructure. In the end, the only USB related lines in a USB example makefile should be `USB_VID = 1209` / `USB_PID = 7D00` (or `${RIOT_TESTING_[VP]ID}`) and the appropriate USEMODULE. Boards that get USB by default (eg. because their default stdio is CDC-ACM, or because gnrc_netdev_default pulls in CEC-ECM) already don't need anything that says "USB" in their Makefiles with this PR.

### CODE REVIEW

File: sys/include/usb.h [hunk 1]
```diff
@@ -29,18 +29,25 @@ extern "C" {
...
  * @{
  */
+
+#if !(defined(CONFIG_USB_VID) && defined(CONFIG_USB_PID))
+#ifdef USB_H_USER_IS_RIOT
+/* Reserved for RIOT standard peripherals as per http://pid.codes/1209/7D00/ */
+#define CONFIG_USB_VID 0x1209
+#define CONFIG_USB_PID 0x7D00
+#else
+#error Please configure your vendor and product IDs. For development, you may set CONFIG_USB_VID=0x1209 CONFIG_USB_PID=0x7D01.
```
Comment (miri64) [inline sys/include/usb.h]: Line-length

File: sys/include/usb.h [hunk 2]
```diff
@@ -29,18 +29,25 @@ extern "C" {
  * @ingroup config
  * @{
  */
+
+#if !(defined(CONFIG_USB_VID) && defined(CONFIG_USB_PID))
+#ifdef USB_H_USER_IS_RIOT
+/* Reserved for RIOT standard peripherals as per http://pid.codes/1209/7D00/ */
+#define CONFIG_USB_VID 0x1209
```
Comment (miri64) [inline sys/include/usb.h]: Parens missing
```suggestion
#define CONFIG_USB_VID (0x1209)
```

**Comment by chrysn** [comment]
> Also, are there other users for usb.h than RIOT?

The USB subsystem is public API, so there can be, and #12268 adds an example that does access those as an application. bergzand's work on USB audio has not hit the repository AFAICT, but is another likely scenario.

> Not sure about the USB_H_USER_IS_RIOT... It leaves a lot of open potential for errors

I'm not trying to make this malice-proof; it's all C after all and people can define what they like. The main goal here is to make the user experience smooth for unaware USB users (no blinky-blinky-get-your-own-code just because my board happens to use USB UART) without letting the RIOT VID/PID pair degenerate into "could be anything really as long as it's RIOT based" codes that more resemble the test pairs than something recognizable.

For that, USB code blocksenabled by USEMODULE (etc, but especially the choice of a board) need to distinguish themselves from USB code blocks introduced by the user. A define looks like the right choice for me to make that distinction; naming is flexible and could just as well be `RIOT_BUILTIN`.

> review comments

Thanks, will work them in with the next batch.

**Comment by miri64** [comment]
> The USB subsystem is public API, so there can be, and #12268 adds an example that does access those as an application. bergzand's work on USB audio has not hit the repository AFAICT, but is another likely scenario.

Ah, so "RIOT" in this case means system internals, not RIOT applications?

**Comment by chrysn** [comment]
Yes. Code that is part of the operating system. So maybe `RIOT_SYSTEM_INTERNAL_USB`?

**Comment by miri64** [comment]
> Yes. Code that is part of the operating system. So maybe `RIOT_SYSTEM_INTERNAL_USB`?

`USB_H_USER_IS_RIOT_INTERNAL`?

**Comment by chrysn** [comment]
Sounds good, will be in next patch series along with fixes for my stylistic missteps.

**Comment by chrysn** [comment]
Updated, reflecting the style nits and the guard rename.

If I may squash, I'll try to make the rest of Travis happy (currently it's hanging at my commit messages, won't get past that without the destructive steps).

**Comment by miri64** [comment]
Yepp, feel free to squash :-)

**Comment by miri64** [comment]
(and change the commit message length)

**Comment by miri64** [comment]
There is also a typo in this commit's message body

**Comment by chrysn** [comment]
Thanks.

The removal of the config code is something that I'd like to do in a follow-up, but in one go with unifying `usb_id_check` into the generic make files. (In the end, only tests/usbus and examples like #12268 should have two lines of PID/VID setting).

The error case around mock is exactly as intended, yes; that does "custom USB stuff" and will need to retain explicit (testing) IDs.

**Pull request merged**